File: Dot examples/ASP Book Examples/Book Examples/Chapter18/VB/Entities/App_Code/Movie.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Data.Linq

Partial Public Class Movie

    Public Shared Function [Select]() As IEnumerable(Of Movie)
        Dim db As New MyDatabaseDataContext()
        Return db.Movies
    End Function

    Public Shared Function SelectByBoxOfficeTotals() As IEnumerable(Of Movie)
        Return [Select]().OrderBy(Function(m) m.BoxOfficeTotals)
    End Function

End Class

File: Dot examples/ASP Book Examples/Book Examples/Chapter18/CS/Entities/App_Code/Movie.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Data.Linq;

public partial class Movie
{
    public static IEnumerable<Movie> Select()
    {
        MyDatabaseDataContext db = new MyDatabaseDataContext();
        return db.Movies;
    }

    public static IEnumerable<Movie> SelectByBoxOfficeTotals()
    {
        return Select().OrderBy(m => m.BoxOfficeTotals);
    }
}
